frontends/infinite-corridor/src/components/forms/EditChannelAppearanceForm.test.tsx

```diff
@@ -0,0 +1,73 @@
+import { act } from "@testing-library/react"
+import {
+  renderTestApp,
+  screen,
+  setMockResponse,
+  fireEvent
+} from "../../test-utils"
+import * as factory from "../../api/fields/factories"
+import { urls } from "../../api/fields"
+
+describe("EditChannelAppearanceForm", () => {
+  it("Displays the field title, appearance inputs", async () => {
+    const field = factory.makeField({ is_moderator: true })
+    setMockResponse.get(urls.fieldDetails(field.name), field)
+    expect(field.is_moderator).toBeTruthy()
+    renderTestApp({ url: `${urls.fieldDetails(field.name)}manage/#appearance` })
+    const descInput = await screen.findByLabelText("Description")
+    const titleInput = await screen.findByLabelText("Title")
+    expect(titleInput.value).toEqual(field.title)
+    expect(descInput.value).toEqual(field.public_description)
+  })
+
+  it("Shows an error if a required field is blank", async () => {
+    const field = factory.makeField({ is_moderator: true })
+    setMockResponse.get(urls.fieldDetails(field.name), field)
+    renderTestApp({ url: `${urls.fieldDetails(field.name)}manage/#appearance` })
+    const titleInput = await screen.findByLabelText("Title")
+    let titleError = screen.queryByText("Title is required")
+    expect(titleError).toBeNull()
+    fireEvent.change(titleInput, {
+      target: { value: "" }
+    })
+    fireEvent.blur(titleInput)
+    titleError = await screen.findByText("Title is required")
+    expect(titleError).toBeInTheDocument()
+  })
+
+  it("updates field values on form submission", async () => {
+    const field = factory.makeField({ is_moderator: true })
+    const newTitle = "New Title"
+    const newDesc = "New Description"
+    // Initial field values
+    setMockResponse.get(urls.fieldDetails(field.name), field)
+    const updatedValues = {
+      ...field,
+      title:              newTitle,
+      public_description: newDesc
+    }
+    setMockResponse.patch(urls.fieldDetails(field.name), updatedValues)
+    renderTestApp({ url: `${urls.fieldDetails(field.name)}manage/#appearance` })
+    const titleInput = await screen.findByLabelText("Title")
+    const descInput = await screen.findByLabelText("Description")
+    const submitBtn = await screen.findByText("Save")
+    fireEvent.change(titleInput, {
+      target: { value: newTitle }
+    })
+    fireEvent.change(descInput, {
+      target: { value: newDesc }
+    })
+    // Expected field values after submit
+    setMockResponse.get(urls.fieldDetails(field.name), updatedValues)
+    // an error will occur without act:
+    // "Warning: An update to Formik inside a test was not wrapped in act(...)."
+    // eslint-disable-next-line testing-library/no-unnecessary-act
+    await act(async () => {
+      fireEvent.submit(submitBtn)
+    })
+    ;[newTitle, newDesc].forEach(newText => {
+      const textHTML = screen.getByText(newText)
+      expect(textHTML).toBeInTheDocument()
+    })
```

Comment:
The `act` warnings are annoying and can be hard to figure out the cause.

In general, testing-library wraps its helpers in `act`, so you don't usuall need them. But it can be hard to figure out what's going on. This is probably the aspect of react testing that I find most frustrating.

The original code was:
```ts
fireEvent.submit(submitBtn)                    // 1. synchronously fire event
;[newTitle, newDesc].forEach(newText => {      
  const textHTML = screen.getByText(newText)   // 2. synchronously find element
  expect(textHTML).toBeInTheDocument()
})
```
Which errors with "Unable to find element". I **think** what was happening was:
1. synchronously fire event. But event handlers are asynchronous, so they won't respond right away
2. synchronously find element. But the event handlers haven't responded yet, so the test fails.

Instead, I would suggest using `findByText`. The `findBy` queries are asynchronous and timeout after a certain time. **So if an element might not exist yet, but should exist soon, use `findBy`.**

This works:
```ts
fireEvent.submit(submitBtn)
await screen.findByText(newTitle)
await screen.findByText(newDesc)
```